packages/injectable/core/index.d.ts

```diff
@@ -90,35 +90,16 @@ export interface Injectable<
   readonly scope?: boolean;
 }
 
-type InjectableLifecycle<InstantiationParam> = InstantiationParam extends void
-  ? {
-      lifecycle?: ILifecycle<void>;
-    }
-  : {
-      lifecycle: ILifecycle<InstantiationParam>;
-    };
-
-type GetInjectableOptions<
-  InjectionInstance extends InjectionTokenInstance,
-  InjectionTokenInstance,
-  InstantiationParam,
-> = InjectableLifecycle<InstantiationParam> &
-  Omit<
-    Injectable<InjectionInstance, InjectionTokenInstance, InstantiationParam>,
-    'lifecycle'
-  >;
+type GetVoidInjectableOptions<I extends TI, TI> = Omit<Injectable<I, TI, void>, "lifecycle"> & {
```

Comment:
`GetInjectableOptionsWithoutInstantiationParameter`

Also: should be accompanied with symmetrical:

`GetInjectableOptionsWithInstantiationParameter`